Lib/test/test_typing.py

```diff
@@ -4346,6 +4347,54 @@ def test_valid_uses(self):
         self.assertEqual(C4.__parameters__, (T, P))
 
 
+class TypeGuardTests(BaseTestCase):
+    def test_basics(self):
+        TypeGuard[int]  # OK
+
+        def foo(arg) -> TypeGuard[int]: ...
+        self.assertEqual(gth(foo), {'return': TypeGuard[int]})
+
+        with self.assertRaises(TypeError):
+            TypeGuard[1]
+        with self.assertRaises(TypeError):
+            TypeGuard[int, str]
+
+    def test_repr(self):
+        self.assertEqual(repr(TypeGuard), 'typing.TypeGuard')
+        cv = TypeGuard[int]
+        self.assertEqual(repr(cv), 'typing.TypeGuard[int]')
+        cv = TypeGuard[Employee]
+        self.assertEqual(repr(cv), 'typing.TypeGuard[%s.Employee]' % __name__)
+        cv = TypeGuard[tuple[int]]
+        self.assertEqual(repr(cv), 'typing.TypeGuard[tuple[int]]')
+
+    def test_cannot_subclass(self):
+        with self.assertRaises(TypeError):
+            class C(type(TypeGuard)):
+                pass
+        with self.assertRaises(TypeError):
+            class C(type(TypeGuard[int])):
+                pass
+
+    def test_cannot_init(self):
+        with self.assertRaises(TypeError):
+            TypeGuard()
+        with self.assertRaises(TypeError):
+            type(TypeGuard)()
+        with self.assertRaises(TypeError):
+            type(TypeGuard[Optional[int]])()
+
+    def test_no_isinstance(self):
+        with self.assertRaises(TypeError):
+            isinstance(1, TypeGuard[int])
+        with self.assertRaises(TypeError):
+            issubclass(int, TypeGuard)
+
```

Comment:
Hmm honestly do we need this? I copied them from the tests for ``Final``, but I noticed that they should apply to all ``@_SpecialForm``. In an ideal world, the entire ``test_typing`` test suite could be greatly simplified it we just had a single test for ``_SpecialForm``

Reply:
I think they're good to have because `_SpecialForm` is an implementation detail, and these tests should test the public API of the module.

In the same vein, I think it'd be useful to test that `get_origin()` and `get_args()` work correctly with `TypeGuard` objects.

Reply:
Agreed. However, I'd maybe cool it a little with the tests for error cases. If in the future we decided to somehow expand the syntax for TypeGuard to allow e.g. literals (`1` could be short for `Literal[1]`?) or multiple arguments, we'd just have to change the code here. I think we went over that for some other novel additions.